### PR TITLE
Fixed HTTPHeaderDict import error.

### PR DESCRIPTION
**Pull request by hramezani**
Fixes https://github.com/getsentry/responses/issues/354

### CODE REVIEW

**Comment by epwalsh** [comment]
This would probably break for older versions of urllib3 then, right? Might need to pin urllib3 to `>=1.26.0` in `setup.py`.

**Comment by hramezani** [comment]
> This would probably break for older versions of urllib3 then, right? Might need to pin urllib3 to `>=1.26.0` in `setup.py`.

Thanks, @epwalsh, I pinned it to `>=1.26.0`

File: responses.py [hunk 1]
```diff
@@ -28,7 +28,7 @@
 try:
     from requests.packages.urllib3.connection import HTTPHeaderDict
 except ImportError:
-    from urllib3.connection import HTTPHeaderDict
+    from urllib3._collections import HTTPHeaderDict
```
Comment (markstory) [inline responses.py:31]: I don't know about this. My understanding is that modules with `_` prefixes are internal and shouldn't be used by consumers of a package. I'm going to open an issue in urllib3 about the removed import causing a compatibility break.

Comment (hramezani) [reply responses.py]: I changed the patch to import `HTTPHeaderDict` from  `urllib3.response` based on https://github.com/urllib3/urllib3/issues/2045#issuecomment-725497199

**Review by pquentin — APPROVED** [review]
Thanks for this, @hramezani! Fetching from `urllib3.response` is going to work on all versions of urllib3 that actually have `HTTPHeaderDict`

**Pull request merged**